### ISSUE TITLE
Consider logging missing `code_verifier` when `code_challenge` is included in authorization request

### ISSUE THREAD

**Issue by sjohnr** [spring-projects/spring-authorization-server#1248]
https://github.com/spring-projects/spring-authorization-server/blob/27a893fbeea7ec1f8ebb85be9242f17b24223e72/oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/CodeVerifierAuthenticator.java#L133-L134

We should consider adding a log entry at DEBUG level in `CodeVerifierAuthenticator` for this case. This would allow the logging level to be tuned specifically for this logging.

**Comment by sjohnr** [comment]
@dciarniello we want to be very conservative about what shows up at INFO or WARN because these logs would show up for most deployments by default, which potentially could be very noisy. I'd rather be asked to turn up logging then turn it down. Instead, these logs are in their own classes at DEBUG so you can opt-in to whichever you want by enabling DEBUG for that particular class (logger) only. Does that make sense?

**Comment by dciarniello** [comment]
@sjohnr , Yes, the reasoning does make sense as there's issue of having enough information in the logs to troubleshoot issue versus having logs that are so noisy that it actually impedes troubleshooting.  In this particular case, however, I would assume that the errors would be infrequent enough that they would not generate a lot of noise so I would lean towards logging them at a higher level.

DEBUG is not something that we would generally turn on in production exactly because of the potential for increased noise.  Having said that, I have not examined the code to see if turning on DEBUG for the classes in question would be excessively noisy.

**Comment by dciarniello** [comment]
Hi, @sjohnr 
100,000 clients isn't something we deal with so I hadn't considered that :-)

I've now looked at the code and I believe that the change will, indeed, work for us.  I'll let you know if it doesn't after we've had a chance to try it out.

Thanks!
